skill/util.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2021, Mycroft AI Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Utility functions for the time skill."""
+from datetime import datetime, timedelta
+from typing import Optional, Tuple
+
+from mycroft.api import GeolocationApi
+from mycroft.util.format import nice_time
+from mycroft.util.parse import extract_duration, extract_datetime
+
+
+class LocationNotFoundError(Exception):
+    """Raise this exception when the geolocation API returns no information."""
+
+    pass
+
+
+def get_geolocation(location: str) -> dict:
+    """Retrieve the geolocation information about the requested location.
+
+    Args:
+        location: a location specified in the utterance
+
+    Returns:
+        A deserialized JSON object containing geolocation information for the
+        specified city.
+
+    Raises:
+        LocationNotFound error if the API returns no results.
+    """
+    geolocation_api = GeolocationApi()
+    geolocation = geolocation_api.get_geolocation(location)
+
+    if geolocation is None:
+        raise LocationNotFoundError("Location {} is unknown".format(location))
+
+    return geolocation
+
+
+def get_duration_from_utterance(utterance: str) -> Optional[timedelta]:
```

Comment:
What is the purpose of this method? It looks like it just wraps an existing method in another file. What is the value ad of this method?

Reply:
it's a convenience function that handles the case of `extract_duration` returning a single value (None) or two values (duration and remaining utterance).